astropy/wcs/wcsapi/tests/test_sliced_low_level_wcs.py

```diff
@@ -0,0 +1,436 @@
+import pytest
+
+import numpy as np
+
+from numpy.testing import assert_equal, assert_allclose
+from astropy.wcs import WCS
+from astropy.io.fits import Header
+from astropy.coordinates import SkyCoord, Galactic
+from astropy.units import Quantity
+from astropy.wcs.wcsapi.sliced_low_level_wcs import SlicedLowLevelWCS, sanitize_slices
+import astropy.units as u
+
+# To test the slicing we start off from standard FITS WCS
+# objects since those implement the low-level API. We create
+# a WCS for a spectral cube with axes in non-standard order
+# and with correlated celestial axes and an uncorrelated
+# spectral axis.
+
+HEADER_SPECTRAL_CUBE = """
+NAXIS   = 3
+NAXIS1  = 10
+NAXIS2  = 20
+NAXIS3  = 30
+CTYPE1  = GLAT-CAR
+CTYPE2  = FREQ
+CTYPE3  = GLON-CAR
+CRVAL1  = 10
+CRVAL2  = 20
+CRVAL3  = 25
+CRPIX1  = 30
+CRPIX2  = 40
+CRPIX3  = 45
+CDELT1  = -0.1
+CDELT2  =  0.5
+CDELT3  =  0.1
+CUNIT1  = deg
+CUNIT2  = Hz
+CUNIT3  = deg
+"""
+
+WCS_SPECTRAL_CUBE = WCS(Header.fromstring(HEADER_SPECTRAL_CUBE, sep='\n'))
+WCS_SPECTRAL_CUBE.pixel_bounds = [(-1, 11), (-2, 18), (5, 15)]
+
+
+@pytest.mark.parametrize("item, ndim, expected", (
+    ([Ellipsis, 10], 4, [slice(None)] * 3 + [10]),
+    ([10, slice(20, 30)], 5, [10, slice(20, 30)] + [slice(None)] * 3),
+    ([10, Ellipsis, 8], 10, [10] + [slice(None)] * 8 + [8])
+))
+def test_sanitize_slice(item, ndim, expected):
+    new_item = sanitize_slices(item, ndim)
+    # FIXME: do we still need the first two since the third assert
+    # should cover it all?
+    assert len(new_item) == ndim
+    assert all(isinstance(i, (slice, int)) for i in new_item)
+    assert new_item == expected
+
+
+EXPECTED_ELLIPSIS_REPR = """
+SlicedLowLevelWCS Transformation
+
+This transformation has 3 pixel and 3 world dimensions
```

Comment:
Is there any chance we could make this repr more concise?  Specifically, put some of these tables on the same rows?  A big annoyance when dealing with WCS reprs is having important information scroll (way) off the screen.

Reply:
The issue is that the output would then be too wide and the tables would get messed up with wrapping. One alternative would be to keep __str__ as the verbose one and have __repr__ be much more concise. What do you think?

paging @Cadair for opinions

Reply:
yeah, if this can't be done while keeping the tables to <80 characters, then just punt.

Reply:
I'll investigate once this is merged - changing the __str__/__repr__ is kind of like changing docs, I think we can do it after feature freeze...

Reply:
I took a look at this but I'm worried that there are users (particularly in the notebook context) who depend on this repr to show them most of the information, and we are subtlety breaking backwards compatibility.  So I'm tempted to say we punt this to a separate issue?  (which we may or may not do for 3.2)?